### PR TITLE
#555 - document Artifactory API

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #555 I've started to document supported Artifactory API, please, take a look and let me know if chosen format is appropriate. You can check readme [here](https://github.com/olenagerasimova/artipie/blob/555-setup/ARTIFACTORY_API_SUPPORT.md).

### CODE REVIEW

File: ARTIFACTORY_API_SUPPORT.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Field name | Type | Meaning | Required
+------------ | ------------- | ------------ | ---------
+key | string | New repository name | Y
+rclass | string | Repository class, only `local` type is supported | Y
+packageType | string | Artifact type, only `docker` type is supported | Y
+dockerApiVersion | string | Docker API version, we only support `V2` | Y
+
+Possible responses:
+- `200 OK` when new repository was successfully created
+- `400 BAD REQUIEST` in the cases when repository with such name already exists or invalid json was sent
```
Comment (olegmoz) [inline ARTIFACTORY_API_SUPPORT.md:22]: @olenagerasimova typo in `REQUIEST`

File: ARTIFACTORY_API_SUPPORT.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- `500 INTERNAL ERROR` in the case of unexpected server error
+
+## Users
+
+### Get the users list
```
Comment (olegmoz) [inline ARTIFACTORY_API_SUPPORT.md:27]: @olenagerasimova title should be capitalized to be in line with other titles in the file and general formatting rules for English texts

File: ARTIFACTORY_API_SUPPORT.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+Returns json with the following fields:
+
+Field name | Type | Meaning | Required
+------ | ------ | ------ | ------
+name | string | User name | Y
+email | string | User email, always constructed as {userName}@artipie.com | Y
+lastLoggedIn | string | Default `2020-01-01T01:01:01.000+01:00` | Y
+realm | string | User realm, value `Internal` is always returned | Y
+
+If user is not found `404 NOT FOUND` status is returned.
```
Comment (olegmoz) [inline ARTIFACTORY_API_SUPPORT.md:64]: @olenagerasimova we have got `Internal` and `internal` values in the docs. Is it a typo or these values are actually different?

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, the format looks good, but I have some suggestions:
* Add links to original Artifactory API docs to get some additional info and compare, for example for here is the link to for Get Users endpoint: https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-GetUsers
* Add small section to main `README.md` about Artifactory API support with link to this file

Also please see some comments about small issues with the text

File: ARTIFACTORY_API_SUPPORT.md (hunk 3 above)
Comment (olenagerasimova) [reply ARTIFACTORY_API_SUPPORT.md:64]: @olegmoz hm, good question. I've copied this example from [artifactory](https://www.jfrog.com/confluence/display/rtf/artifactory+rest+api#ArtifactoryRESTAPI-GetUsers), there they use `internal`, but [this](https://www.jfrog.com/confluence/display/JFROG/Security+Configuration+JSON#SecurityConfigurationJSON-application/vnd.org.jfrog.artifactory.security.User+json) example have `Internal`.
In artipie we actually always use `Internal` with the capital I.

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected and answered

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#557` is not assigned, can't get performer

**Pull request merged**